pkg/api/validate/dynamic/quota.go

```diff
@@ -133,13 +83,14 @@ func (dv *dynamic) ValidateQuota(ctx context.Context, oc *api.OpenShiftCluster)
 	}
 
 	requiredResources := map[string]int{}
-	err := addRequiredResources(requiredResources, oc.Properties.MasterProfile.VMSize, 3)
+	err := addRequiredResources(requiredResources, oc.Properties.MasterProfile.VMSize, "MasterProfile", 3)
 	if err != nil {
 		return err
 	}
+
 	//worker node resource calculation
 	for _, w := range oc.Properties.WorkerProfiles {
-		err = addRequiredResources(requiredResources, w.VMSize, w.Count)
+		err = addRequiredResources(requiredResources, w.VMSize, "WorkerProfile", w.Count)
```

Comment:
As I mentioned ealier - this error is not for customers and it is not a big deal if we just have `unexpected node VMSize %s"`: the issue will be very clear - we miss vm type in our code.

But if we want to have more specificity, we should probably call the argument `path` instead of `role` and do `fmt.Sprintf("properties.workerProfiles[%d].vmSize", i)` for workers and `"properties.masterProfile.vmSize"` for master.

This is what we do for customer facing errors: we include path to the field which has an invalid value.

But I would just drop it. No need for more specificity here, IMO.